src/common/snippets/src/lowered/pass/insert_buffers.cpp

```diff
@@ -28,6 +29,39 @@ std::vector<size_t> get_buffer_loop_ids(const std::vector<size_t>& lhs, const st
     }
     return buffer_loop_ids;
 }
+
+ov::Shape compute_allocation_shape(const LinearIR::LoopManagerPtr& loop_manager,
+                                   const std::vector<size_t>& buffer_loop_ids,
+                                   const std::vector<size_t>& parent_loop_ids,
+                                   const ov::Output<ov::Node>& parent_output,
+                                   const int allocation_rank) {
+    const size_t rank = allocation_rank >= 0 ? allocation_rank : parent_output.get_shape().size();
+    ov::Shape allocation_shape(rank);
+    const auto port = lowered::PortDescriptorUtils::get_port_descriptor_ptr(parent_output);
+    const auto planar_shape = utils::get_reordered_planar_shape(ov::Shape{port->get_shape()}, port->get_layout());
+    for (size_t i = 0; i < rank; ++i) {
+        allocation_shape[rank - 1 - i] = planar_shape[planar_shape.size() - 1 - i].get_length();
+    }
+
+    if (buffer_loop_ids.empty() || parent_loop_ids.empty()) {
+        return allocation_shape;
+    }
+
+    // In some cases it's possible to allocate less shape
+    // 1. Buffer and its parent are in the same loop: allocation size for the outer dimension can be extracted from loop increment
+    // 2. Buffer is outside the parent's loops: allocation size can be extracted from the corresponding loop work amount
+    // TODO: Use general logic with the help of memory counts for allocation shape computation
+    if (buffer_loop_ids.back() == parent_loop_ids.back()) {
+        const auto buffer_loop = loop_manager->get_loop_info(buffer_loop_ids.back());
+        allocation_shape[0] = buffer_loop->increment;
+    } else {
+        for (size_t i = 0; i < std::min(rank, parent_loop_ids.size()); ++i) {
+            const auto loop = loop_manager->get_loop_info(parent_loop_ids[parent_loop_ids.size() - 1 - i]);
```

Comment:
Minor (for any taste)
``` suggestion
            const auto loop = loop_manager->get_loop_info(*(parent_loop_ids.rbegin() + i));
```

Reply:
Your option is better) Changed, thank you